### PR TITLE
set literal conversions

### PR DESCRIPTION
**Pull request by pq**
Still WIP and not ready for landing until and SDK with Set literals lands in Flutter.

Feedback welcome in the meantime.  (Especially if there are cases I mis-migrated.)

/cc @Hixie @amirh 

/fyi @bwilkerson

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/material/chip_demo.dart
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/material/chip_demo.dart]: can this be const?

**Comment by Hixie** [comment]
Many of these literals seem like they should be const now.

It would be good to enable the lint at the same time, so we don't regress this.

Notwithstanding the above, LGTM.

**Comment by pq** [comment]
FYI: holding off until Set-literals have landed internally.

Also note that the next SDK pull should get a linter that has `prefer_const_declarations` fixed to not produce bogus advice for (wrongly) implicitly constant Set literals.  That should clear up remaining `analyze` bot failures.

File: dev/bots/analyze.dart
Comment (yetanothercoder) [inline dev/bots/analyze.dart]: can't second type specification be omitted?
like: `final Set<String> outOfDate = {};`

Comment (a14n) [reply dev/bots/analyze.dart]: yes but the internal Flutter style guide requires to always specify types.

**Comment by goderbauer** [comment]
@pq What's the status on this one? I believe set literals have rolled internally?

**Comment by pq** [comment]
@goderbauer: yep.  I think this is safe % rebase (&& perf).

(I'll slate some time to tidy this up after 11:59 PM PT 😄 )

**Comment by pq** [comment]
@Hixie: I took a crack at migrating to `const` where I saw the opportunity.  Feel free to double-check.

@goderbauer: this looks good to me locally but will wait for the build to go through (which is currently broken).

**Review by goderbauer — APPROVED** [review]
LGTM!

Thanks for doing the migration!

**Pull request merged**
